src/core_modules/capture-core/metaDataStoreLoaders/IOUtils/IOUtils.types.js

```diff
@@ -0,0 +1,28 @@
+// @flow
+export type ApiQuery = {
+    resource: string,
+    id?: string,
+    data?: Object,
+    params?: Object,
+};
+
+export type ApiQueryExtended = {
+    resource: string,
+    id?: string,
+    data?: Object | (variables: Object) => Object,
+    params?: Object | (variables: Object) => Object,
+};
+
+export type StoreName = string;
+
+export type ConvertFn = (apiResponse: any) => any;
+
+export type QuickStoreOptions = {
+    onConvert?: ConvertFn,
```

Comment:
Because a conversion isn't always needed? I think it's fine either way here.

Reply:
I am saying this because I haven't seen cases where we dont pass it. And I dont know well the whole application so there might be some cases that I simply dont know. 

Cant we start by having it required and change it if it ends up being optional for some cases then we change it? This way when someone reads the function can understand that they need to pass a convert function in.

Reply:
No offence, but this is very subjective. I feel the function very well can be used without supplying the converter. As the PR author I'm going to say we stick with the the current implementation.

Reply:
It can be used without the converter I totally agree. The point is that we _never_ have used it this way till now that's why I am asking what is the point of having the argument as optional. 

If its optional in the types but we never use the function without passing the `onConvert` parameter then having it optional is confusing the reader.

Reply:
Hey, resolving this :)

Reply:
Talked it through and made some compromises here and there. Going for convertQueryResponse (onConvert) as mandatory.